nowcasting_dataloader/datasets.py

```diff
@@ -179,6 +187,182 @@ def __getitem__(self, batch_idx: int) -> dict:
+        if len(batch["hrvsatellite"].get("data", [])) > 0:
+            past_hrv_satellite_data = batch["hrvsatellite"]["data"][
+                :, :, : self.current_timestep_index
+            ]
+            x["hrvsatellite"] = past_hrv_satellite_data
+        if len(batch["pv"].get(PV_YIELD, [])) > 0:
+            past_pv_data = batch["pv"][PV_YIELD][:, :, : self.current_timestep_index]
+            x[PV_YIELD] = past_pv_data
+            x[PV_SYSTEM_ID] = batch["pv"][PV_SYSTEM_ID]
+        if len(batch["nwp"].get("data", [])) > 0:
+            # We can give future NWP too, as that will be available
+            x[NWP_DATA] = batch["nwp"]["data"]
+        if len(batch["topographic"].get(TOPOGRAPHIC_DATA, [])) > 0:
+            # Need to expand dims to get a single channel one
+            # Results in topographic maps with [Batch, Channel, H, W]
+            x[TOPOGRAPHIC_DATA] = batch["topographic"][TOPOGRAPHIC_DATA]
+
+        # Only GSP information we give to the model to train on is the IDs and physical locations
+        x[GSP_ID] = batch["gsp"][GSP_ID]
+
+        # Now creating the target data
+        target[GSP_YIELD] = batch["gsp"][GSP_YIELD][:, :, self.current_timestep_index_30 :]
+        target[GSP_ID] = batch["gsp"][GSP_ID]
+
+        if self.add_satellite_target:
+            future_sat_data = batch["satellite"]["data"][:, :, self.current_timestep_index :]
+            target[SATELLITE_DATA] = future_sat_data
+        if self.add_hrv_satellite_target:
+            future_hrv_sat_data = batch["hrvsatellite"]["data"][:, :, self.current_timestep_index :]
+            target["hrv_" + SATELLITE_DATA] = future_hrv_sat_data
+
+        # Add position encodings
+        if self.add_position_encoding:
+            if len(x.get("satellite", [])) > 0:
+                x = self.add_encodings(
+                    x, "satellite", batch, self.current_timestep_index, self.add_satellite_target
+                )
+                if self.add_hrv_satellite_target:
+                    x[SATELLITE_DATA + "_query"] = x.pop("satellite_query")
+            if len(x.get("hrvsatellite", [])) > 0:
+                x = self.add_encodings(
+                    x,
+                    "hrvsatellite",
+                    batch,
+                    self.current_timestep_index,
+                    self.add_hrv_satellite_target,
+                )
+                if self.add_hrv_satellite_target:
+                    x["hrv_" + SATELLITE_DATA + "_query"] = x.pop("hrvsatellite_query")
+            if len(x.get(TOPOGRAPHIC_DATA, [])) > 0:
+                x = self.add_encodings(x, TOPOGRAPHIC_DATA, batch, 0, False)
+            if len(x.get(NWP_DATA, [])) > 0:
+                x[NWP_DATA] = torch.cat(
+                    [x[NWP_DATA], batch[NWP_DATA + "_position_encoding"]], dim=1
+                )
+            if len(x.get(PV_YIELD, [])) > 0:
+                x = self.add_encodings(x, PV_YIELD, batch, self.current_timestep_index, False)
```

Comment:
Can we also add an encoding of the geographical location of each PV system?  (Maybe as a separate PR?) 

(not super-super urgent as the model might be able to infer to location of each PV system)

(Sorry if this code already does this and I've missed it!)

Reply:
This already does it! It takes the coordinates of the GSP and PV systems and adds them in NetCDFDataset position encodings.